File: src/main/kotlin/com/huawei/cangjie1/psi/CjEnumEntry.java
package com.huawei.cangjie1.psi;

import com.huawei.cangjie1.name.ClassId;
import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiClass;
import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiEnumConstant;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Collections;
import java.util.List;


//public class CjEnumEntry extends CjClass {
//    public CjEnumEntry(@NotNull ASTNode node) {
//        super(node);
//    }
//
//    public CjEnumEntry(@NotNull CangJieClassStub stub) {
//        super(stub);
//    }
//
//    @NotNull
//    @Override
//    public List<CjSuperTypeListEntry> getSuperTypeListEntries() {
//        CjInitializerList initializerList = getInitializerList();
//        if (initializerList == null) {
//            return Collections.emptyList();
//        }
//        return initializerList.getInitializers();
//    }
//
//    public boolean hasInitializer() {
//        return !getSuperTypeListEntries().isEmpty();
//    }
//
//    @Nullable
//    @Override
//    public ClassId getClassId() {
//        return null;
//    }
//
//    @Nullable
//    public CjInitializerList getInitializerList() {
//        return getStubOrPsiChild(CjStubElementTypes.INITIALIZER_LIST);
//    }
//
//    @Override
//    public <R, D> R accept(@NotNull CjVisitor<R, D> visitor, D data) {
//        return visitor.visitEnumEntry(this, data);
//    }
//
//    @Override
//    public boolean isEquivalentTo(@Nullable PsiElement another) {
//        if (another instanceof PsiEnumConstant) {
//            PsiEnumConstant enumConstant = (PsiEnumConstant) another;
//            PsiClass containingClass = enumConstant.getContainingClass();
//            if (containingClass != null) {
//                String containingClassQName = containingClass.getQualifiedName();
//                if (containingClassQName != null && enumConstant.getName() != null) {
//                    String theirFQName = containingClassQName + "." + enumConstant.getName();
//                    if (theirFQName.equals(getQualifiedName())) {
//                        return true;
//                    }
//                }
//            }
//        }
//        return super.isEquivalentTo(another);
//    }
//}

File: src/main/kotlin/com/linqingying/cangjie/psi/CjEnumEntry.kt
package com.linqingying.cangjie.psi

import com.linqingying.cangjie.psi.stubs.CangJieEnumEntryStub
import com.linqingying.cangjie.psi.stubs.elements.CjStubElementTypes
import com.intellij.lang.ASTNode

class CjEnumEntry : CjTypeStatement {
    constructor(node: ASTNode) : super(node)

    constructor(
        stub: CangJieEnumEntryStub

    ) : super(stub, CjStubElementTypes.ENUM_ENTRY)

    override fun <R, D> accept(visitor: CjVisitor<R, D>, data: D?): R {
        return visitor.visitEnumEntry(this, data)
    }
    override val typeName: String
        get() = "enum entry"
    override fun hasExplicitPrimaryConstructor(): Boolean {
        return typeReferences.isNotEmpty()
    }

    override fun toString(): String {
        return node.elementType.toString()
    }

    val typeEntry: CjEnumEntryTypeEntry?
        get() = findChildByClass(CjEnumEntryTypeEntry::class.java)
    val typeReferences: List<CjTypeReference>
        get() {
            if (typeEntry != null) {
                return typeEntry!!.getStubOrPsiChildrenAsList(CjStubElementTypes.TYPE_REFERENCE)
            }
            return emptyList()


        }
}
